### ISSUE TITLE
Transposition to flat keys

### ISSUE THREAD

**Issue by fredrikved** [opensheetmusicdisplay/opensheetmusicdisplay#1345]
We're having an issue where our sheets are rendered with a more difficult to read transcription than necessary when transposing the scores to a flat key. For example a sheet in E transposed to Db:

Manually written: 
![image](https://user-images.githubusercontent.com/75886490/224736607-a420667e-f296-469b-b1e0-aaa6f78e30f7.png)

Output from OSDM (+9, with the source originally in E)
![image](https://user-images.githubusercontent.com/75886490/224736919-941ef649-7236-4231-95e0-05a004743d0a.png)

As we can see, E's for example are never notated, even though they also represent the same note as D#. Same with B's, making this a bit more difficult than necessary to read. 

Is it possible to fix this?

**Comment by fredrikved** [comment]
Here's an example file where that issue happens (originally in E, transposing for Eb instruments (+9) results in bad readability).

[ID-98013_Det_er_en_skatt_bak_dekker_gjemt.musicxml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/11049781/ID-98013_Det_er_en_skatt_bak_dekker_gjemt.musicxml.zip)

**Comment by ammatwain** [comment]
Good morning Simon, I uploaded the MusicXML file to my test area at [https://ammatwain.github.io/transport-calculator/](https://ammatwain.github.io/transport-calculator/), and it is transposed correctly.

**Comment by fredrikved** [comment]
I fixed this for our case with a [https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/compare/develop...fredrikved:opensheetmusicdisplay:develop](hardcoded if this then that solution)

Basically, B-flats and E sharps can be written as just that instead of A-sharp and F-neutral, and it only occurred for those notes, so the issues we had are no longer to be seen.

**Comment by ammatwain** [comment]
Yes, Fredrik, my class works with equal temperament. The expected transposed note is delivered by the final link in a chain of key relationships.

**Comment by fredrikved** [comment]
Oh now I see. Used the wrong transposition selector on your page 🙈 

Your solution works great for this application.

**Comment by ammatwain** [comment]
Darn it... I should have put it first!...
